### PR TITLE
Running vsi test

### PR DESCRIPTION
**Pull request by jmarolf**

### CODE REVIEW

File: build/Rulesets/Roslyn.ruleset [hunk 1]
```diff
@@ -126,8 +126,8 @@
     <Rule Id="IDE0019WithoutSuggestion" Action="Error" />
     <Rule Id="IDE0020" Action="Error" />                      <!-- Use pattern matching                         if (control is Button)                              if (control is Button button)         -->
     <Rule Id="IDE0020WithoutSuggestion" Action="Error" />
-    <Rule Id="IDE0028" Action="Error" />                      <!-- Use coalesce expression                      string y = x == null ? string.Empty : x;            string y = x ?? string.Empty;         -->
-    <Rule Id="IDE0028WithoutSuggestion" Action="Error" />      
+    <Rule Id="IDE0028" Action="Info" />                       <!-- Collection initialization can be simplified                                                                                            -->
```
Comment (srivatsn) [inline build/Rulesets/Roslyn.ruleset]: Unrelated change? If so please remove.

File: build/Targets/References.targets [hunk 2]
```diff
@@ -222,7 +222,7 @@
       <ItemGroup>
 
         <PackageReference Include="Moq" Version="4.2.1402.2112" />
-        <PackageReference Include="xunit" Version="2.1.0" />
+        <PackageReference Include="xunit" Version="2.2.0-beta4-build3444" />
```
Comment (srivatsn) [inline build/Targets/References.targets]: Why do we need 2.2.0 and why a beta4 build and 2.2.0 itself? @davkean [had mentioned](https://github.com/dotnet/project-system/pull/1912#issuecomment-291021172) that 2.2.0 regressed perf when he had tried to upgrade.

File: build/build.proj [hunk 3]
```diff
@@ -84,7 +85,7 @@
   <Target Name="Test" Condition="'$(RunTests)' == 'true'">
 
     <ItemGroup>
-      <TestAssembly Include="$(TestsDirectory)*Tests.dll" />
+      <TestAssembly Include="$(TestsDirectory)*UnitTests.dll" />
```
Comment (srivatsn) [inline build/build.proj]: You need to update the version of xunit runner below at line 95 - that's why the tests are failing.,

File: build/build.proj [hunk 4]
```diff
@@ -103,5 +104,47 @@
     <Error Text="There were test failures, for full log see %(HtmlTestFile.FullPath)" Condition="$(ExitCode) != 0" />
 
   </Target>
- 
+
+  <Target Name="TestIntegration" Condition="'$(RunIntegrationTests)' == 'true'">
+    <PropertyGroup>
+      <DownloadRoslynVsixesExe>$(NUGET_PACKAGES)\roslyntools.downloadroslynvsixes\0.3.2-beta\tools\DownloadRoslynVsixes.exe</DownloadRoslynVsixesExe>
```
Comment (srivatsn) [inline build/build.proj]: Can we put the package version numbers in a property so that there's one place to update? Same for xunit as well.

File: netci.groovy [hunk 5]
```diff
@@ -49,168 +49,50 @@ build.cmd /no-node-reuse /no-deploy-extension /${configuration.toLowerCase()}
...
-SET VS150COMNTOOLS=%ProgramFiles(x86)%\\Microsoft Visual Studio\\2017\\Enterprise\\Common7\\Tools\\
-SET VSSDK150Install=%ProgramFiles(x86)%\\Microsoft Visual Studio\\2017\\Enterprise\\MSBuild\\Microsoft\\VisualStudio\\v15.0\\VSSDK\\
-SET VSSDKInstall=%ProgramFiles(x86)%\\Microsoft Visual Studio\\2017\\Enterprise\\MSBuild\\Microsoft\\VisualStudio\\v15.0\\VSSDK\\
-
-build.cmd /no-node-reuse /release /skiptests
-""")
-
-            // Patch all the MSBuild xaml and targets files from the current roslyn-project-system commit into VS install.
-            batchFile("""
-echo *** Patch the MSBuild xaml and targets ***
```
Comment (srivatsn) [inline netci.groovy]: Deleting this code means that the just-built rules\targets wont take effect though right? I had set an env variable in launchSettings.json to help with F5 - those env vars should probably be set here as well

File: netci.groovy [hunk 6]
```diff
@@ -49,168 +49,50 @@ build.cmd /no-node-reuse /no-deploy-extension /${configuration.toLowerCase()}
...
 
-exit /b 0
+        def newVsiJob = job(newVsiJobName) {
+            // This opens the set of build steps that will be run.
+            steps {
+                // Indicates that a batch script should be run with the build string (see above)
+                // Also available is:
+                // shell (for unix scripting)
+                batchFile("""
+echo *** Installing 2.0 CLI ***
```
Comment (srivatsn) [inline netci.groovy]: We are installing the latest CLI so that we get the latest SDK and templates? If so, I think we should install the CLI from the rel/1.0.0 branch for now* because that's the one that's shipping in VS. We first need to make sure the integration of bits in shipping VS is good. We could probably have a different queue that installs the 2.0 CLI and tests as well to make sure that 2.0 changes don't break VS.

* - Btw @livarcocc is figuring out which branch of CLI will get inserted to VS. It's rel/1.0.0 but is going to change soon- we should make sure to update this when that happens.

File: src/Microsoft.VisualStudio.ProjectSystem.IntegrationTests/Utilities/SharedIntegrationHostFixture.cs [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+        public const string MSBuildPackageId = "Microsoft.Component.MSBuild";
+        public const string Net46TargetingPackPackageId = "Microsoft.Net.Component.4.6.TargetingPack";
+        public const string PortableLibraryPackageId = "Microsoft.VisualStudio.Component.PortableLibrary";
+        public const string RoslynCompilerPackageId = "Microsoft.VisualStudio.Component.Roslyn.Compiler";
+        public const string RoslynLanguageServicesPackageId = "Microsoft.VisualStudio.Component.Roslyn.LanguageServices";
+        public const string VsSdkPackageId = "Microsoft.VisualStudio.Component.VSSDK";
+
+        public static readonly ImmutableHashSet<string> RequiredPackageIds = ImmutableHashSet.Create(MSBuildPackageId, Net46TargetingPackPackageId, PortableLibraryPackageId, RoslynCompilerPackageId, RoslynLanguageServicesPackageId, VsSdkPackageId);
+
+        // This class has no code, and is never created. Its purpose is simply
```
Comment (srivatsn) [inline src/Microsoft.VisualStudio.ProjectSystem.IntegrationTests/Utilities/SharedIntegrationHostFixture.cs]: Make this a summary doc comment.

File: src/Microsoft.VisualStudio.ProjectSystem.IntegrationTests/Utilities/FeatureAttribute.cs [hunk 8]
```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.VisualStudio.ProjectSystem.IntegrationTests
+{
+    internal partial class FeatureAttribute
+    {
+        public const string AutomaticEndConstructCorrection = nameof(AutomaticEndConstructCorrection);
```
Comment (srivatsn) [inline src/Microsoft.VisualStudio.ProjectSystem.IntegrationTests/Utilities/FeatureAttribute.cs]: Any reason to keep this in sync with Roslyn? Otherwise we can get rid of a bunch of these I assume.

File: build/build.proj (hunk 4 above)
Comment (basoundr) [inline build/build.proj]: We done this for some of the Nuget packages already [here](https://github.com/dotnet/project-system/blob/master/build/Targets/VSL.Versions.targets#L21)

**Comment by basoundr** [comment]
Also it would be helpful to split this change into commits with appropriate commit message to understand what each change is trying to do.

File: build/Targets/References.targets (hunk 2 above)
Comment (jmarolf) [inline build/Targets/References.targets]: This is the version that [roslyn is using](https://github.com/dotnet/roslyn/blob/2223e15607ec068e17c23dc6aa9718a8366dcb80/build/Targets/Packages.props#L223)  and therefore the version required by the vsi test framework. I could try amd make it so that only the VSI tests depend on this version but it becomes difficult to reason about which package nuget deploys as it tries to unify things.

File: build/Rulesets/Roslyn.ruleset (hunk 1 above)
Comment (jmarolf) [inline build/Rulesets/Roslyn.ruleset]: [this PR](https://github.com/dotnet/roslyn/pull/16679) changed the diagnostic IDs around. To preserve this behavior I will need update this to be `IDE0029`

Comment (srivatsn) [inline build/Rulesets/Roslyn.ruleset]: Oh I see - since you are running integration tests against latest roslyn, this will fail otherwise? Let\s bump to IDE0029 then

File: netci.groovy (hunk 5 above)
Comment (jmarolf) [inline netci.groovy]: ah right, I though for some reason VSIX deploy would do something.  I'll add this back

File: src/Microsoft.VisualStudio.ProjectSystem.IntegrationTests/Utilities/FeatureAttribute.cs (hunk 8 above)
Comment (jmarolf) [inline src/Microsoft.VisualStudio.ProjectSystem.IntegrationTests/Utilities/FeatureAttribute.cs]: These are used to wait for features.  They should be moved into the test framework itself so that things like `VisualStudio.Workspace.WaitForAsyncOperations(FeatureAttribute.SolutionCrawler)` work.  Arguable everything in the Utilities folder should be moved to the test framework itself.  I'll talk to @tannergooding about it.

File: build/build.proj (hunk 3 above)
Comment (jmarolf) [inline build/build.proj]: Done.

File: netci.groovy [hunk 9]
```diff
@@ -49,168 +49,68 @@ build.cmd /no-node-reuse /no-deploy-extension /${configuration.toLowerCase()}
...
-call "%DeveloperCommandPrompt%" || goto :BuildFailed "VsMSBuildCmd.bat"
-
-SET VSSDK150Install=%ProgramFiles(x86)%\\Microsoft Visual Studio\\2017\\Enterprise\\MSBuild\\Microsoft\\VisualStudio\\v15.0\\VSSDK\\
-SET VSSDKInstall=%ProgramFiles(x86)%\\Microsoft Visual Studio\\2017\\Enterprise\\MSBuild\\Microsoft\\VisualStudio\\v15.0\\VSSDK\\
-SET MSBUILDEXE=%ProgramFiles(x86)%\\Microsoft Visual Studio\\2017\\Enterprise\\MSBuild\\15.0\\Bin\\MSBuild.exe
-
-pushd %WORKSPACE%\\sdk
-echo *** Build SDK
-call build.cmd -Configuration release -SkipTests || goto :BuildFailed "SDK"
+xcopy /SIY .\\bin\\${configuration}\\Rules\\*.xaml "%VS_MSBUILD_MANAGED%"
```
Comment (srivatsn) [inline netci.groovy]: Instead of doing the xcopy can we just set the env variables for the targets like we do in launchSettings.json?

File: build/Targets/References.targets (hunk 2 above)
Comment (srivatsn) [inline build/Targets/References.targets]: Hmm.. can you verify that perf for test explorer doesn't regress because of this?

File: build/build.proj (hunk 3 above)
Comment (davkean) [inline build/build.proj]: Why was this change made?

Comment (davkean) [inline build/build.proj]: Ah I see.

File: netci.groovy (hunk 9 above)
Comment (jmarolf) [inline netci.groovy]: yes, that would be simpler

File: build/Targets/References.targets (hunk 2 above)
Comment (jmarolf) [inline build/Targets/References.targets]: I can do some wall-clock timers.  How bad was the regression?  Was it on discovery or test runs?

Comment (srivatsn) [inline build/Targets/References.targets]: don't know - @davkean ?

Comment (jmarolf) [inline build/Targets/References.targets]: running `@powershell -NoProfile -ExecutionPolicy Bypass -Command "Measure-Command { .\build.cmd}"`

I get the following results

| xunit version | total seconds | 
| ------------- |:-------------:| 
| 2.2.0  | 86.6895754 | 
| 2.1.0  | 83.7018841 |

Comment (jmarolf) [inline build/Targets/References.targets]: So a 3.5% regression

Comment (tannergooding) [inline build/Targets/References.targets]: @jmarolf, any reason we don't move to the 2.2.0 RTM version?

Comment (srivatsn) [inline build/Targets/References.targets]: Jon's moved to 2.2.0 in the latest commit. The perf looks fine - lets go ahead with this.
